ba/tutorials/tour/annotations.md

```diff
@@ -0,0 +1,132 @@
+> Pošto Scala nema provjerene izuzetke, Scala metode _moraju_ imati jednu ili više `throws` anotacija kako bi Java kod mogao uhvatiti iste.
+
+    package examples
+    import java.io._
+    class Reader(fname: String) {
+      private val in = new BufferedReader(new FileReader(fname))
+      @throws(classOf[IOException])
+      def read() = in.read()
+    }
+
+Sljedeći Java program prikazuje sadržaj fajla s imenom koje je proslijeđeno kao prvi argument `main` metode.
+
+    package test;
+    import examples.Reader;  // Scala class !!
+    public class AnnotaTest {
+        public static void main(String[] args) {
+            try {
+                Reader in = new Reader(args[0]);
+                int c;
+                while ((c = in.read()) != -1) {
+                    System.out.print((char) c);
+                }
+            } catch (java.io.IOException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+    }
+
+Kada bi zakomentarisali `throws` anotaciju u klasi Reader dobili bi sljedeću grešku s porukom pri kompajliranju Java main programa:
+
+    Main.java:11: exception java.io.IOException is never thrown in body of
+    corresponding try statement
+            } catch (java.io.IOException e) {
+              ^
+    1 error
+
+### Java anotacije ###
+
+**Napomena:** Pobrinite se da koristite `-target:jvm-1.5` opciju sa Java anotacijama.
+
+Java 1.5 je uvela korisnički definisane metapodatke u formi [anotacija](http://java.sun.com/j2se/1.5.0/docs/guide/language/annotations.html). Ključna sposobnost anotacija je da koriste parove ime-vrijednost za inicijalizaciju svojih elemenata. Naprimjer, ako nam treba anotacija da pratimo izvor neke klase mogli bismo je definisati kao
+
+    @interface Source {
+      public String URL();
+      public String mail();
+    }
+
+I upotrijebiti je kao npr:
+
+    @Source(URL = "http://coders.com/",
+            mail = "support@coders.com")
+    public class MyClass extends HisClass ...
+
+Primjena anotacije u Scali izgleda kao poziv konstruktora, dok se za instanciranje Javinih anotacija moraju koristiti imenovani argumenti:
+
+    @Source(URL = "http://coders.com/",
+            mail = "support@coders.com")
+    class MyScalaClass ...
+
+Ova sintaksa je ponekad naporna, npr. ako anotacija ima samo jedan element(bez defaultne vrijednosti), pa po konvenciji, 
+ako se koristi naziv `value` onda se u Javi može koristiti i konstruktor-sintaksa:
+
+    @interface SourceURL {
+        public String value();
+        public String mail() default "";
+    }
+
+I upotrijebiti je kao npr:
+
+    @SourceURL("http://coders.com/")
+    public class MyClass extends HisClass ...
+
+U ovom slučaju, Scala omogućuje istu sintaksu:
+
+    @SourceURL("http://coders.com/")
+    class MyScalaClass ...
+
+Element `mail` je specificiran s defaultnom vrijednošću tako da ne moramo eksplicitno navoditi vrijednost za njega. 
+Međutim, ako trebamo, ne možemo miksati dva Javina stila:
```

Comment:
:) This one is hard. I would not use mixable in the first place; can be mixed together is fine. I 6 years with Scala I have never heard that word.